### ISSUE TITLE
Crash on first Start (1.4.2)

### ISSUE THREAD

**Issue by Mannshoch** [nextcloud/android#1316]
### Actual behaviour
I Installed Nextcloud from f-droid after first start Nextclud crashed.

### Expected behaviour
nextcloud do start
 
### Steps to reproduce
1. install nextcloud 
2. start nextcloud
3. see Question for Storrage access and in background of this pop-up the crash information pop-up.

### Environment data
Android version: 6.0.1

Device model: Fairphone 2

Stock or customized system:  FP Open 17.07.6
[https://code.fairphone.com/projects/fp-osos/user/fairphone-open-source-os-downloads.html](https://code.fairphone.com/projects/fp-osos/user/fairphone-open-source-os-downloads.html)

Nextcloud app version: 1.4.2

### Logs

[Log-next-2017-08-11-02.28.31.txt](https://github.com/nextcloud/android/files/1217996/Log-next-2017-08-11-02.28.31.txt)

```
**NOTE:** Be super sure to remove sensitive data like passwords, note that everybody can look here! You can use the Issue Template application to prefill some of the required information: https://apps.nextcloud.com/apps/issuetemplate

**Comment by jasonbayton** [comment]
I can replicate on the Google play beta 2.0.0 RC

@mario @tobiaskaminsky @andyscherzinger app is unusable and I'm back on one of the dev APKs for now

**Comment by Mannshoch** [comment]
I tested following Version from f-droid:

- 1.4.2 = Crash
- 1.4.1 = Crash
- 1.4.1 RC2 = works without Problem.

**Comment by tobiasKaminsky** [comment]
@jasonbayton can you give us a logcat?
@Mannshoch is this a typo or do you really mean 1.4.1 rc2? Can you also provide us with a logcat?

**Comment by tobiasKaminsky** [comment]
Possible cause in logcat by @Mannshoch 

> InputDispatcher,Android-System,channel 'f18cc21 com.nextcloud.client/com.owncloud.android.ui.activity.FileDisplayActivity (server)' ~ Channel is unrecoverably broken and will be disposed!

**Comment by Mannshoch** [comment]
No that's correct:

![screenshot_20170814-120218](https://user-images.githubusercontent.com/7795891/29267371-b35de67e-80e8-11e7-90e8-efa576e1aa12.png)

**Comment by tobiasKaminsky** [comment]
@Mannshoch do you have some more output of logcat? Though there is an error captured, I doubt that this is the real cause of your problem.

**Comment by tobiasKaminsky** [comment]
@Mannshoch searching for fairphone 2 and the error of your logcat, I get this link:
https://github.com/microg/android_packages_apps_GmsCore/issues/370

Do you use some kind of custom google apps?
Can/want you try the google play store beta release?
 https://play.google.com/apps/testing/com.nextcloud.client

**Comment by Mannshoch** [comment]
At moment I'm not able to get Google apps, but I'll test an actual GMSCore.

**Comment by tobiasKaminsky** [comment]
This might fix it: #1362 
@Mannshoch do you installed oc client in parallel?

**Comment by Mannshoch** [comment]
No.

**Comment by Mannshoch** [comment]
Feedback:

1. Upgrade from 1.4.1 Rc2 to 1.4.2 worked and also nextcloud worked after that.
2. Remove and reinstall Nexcloud -> Crash 

Conclusion: Crash in 1.4.2 only happened if no account already exist

3. Upgraded GMSCore to 0.2.4-103 -> no effect.
4. Uninstalled GMSCore -> no effect

Conclusion -> GMSCore is not the source of this error

**Comment by tobiasKaminsky** [comment]
@Mannshoch can you try it with RC4, found via google play store (as beta tester)?

**Comment by Mannshoch** [comment]
Sry, but I do not have Google Palystore installed

**Comment by tobiasKaminsky** [comment]
@Mannshoch now it is also available on fdroid: https://f-droid.org/packages/com.nextcloud.client/
For the RC5 you have manually go to "versions" and select it.

**Comment by Mannshoch** [comment]
Is Crashing 

[Log-Next--2017-08-31-04.45.43.txt](https://github.com/nextcloud/android/files/1267939/Log-Next--2017-08-31-04.45.43.txt)

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky @mario could this be the originating issue when looking at the log:
```
2017-08-31 04:45:26,INFORMATION,AccountAuthenticator,Unknown,Adding account with type nextcloud and auth token null
```

**Comment by mario** [comment]
@jasonbayton do you have a test account I can test this with?

**Comment by AndyScherzinger** [comment]
To some extends reading through the comments it seems to be an account creation issue, so when an account has been present it worked and for a fresh install the account created was corrupt.

**Comment by mario** [comment]
KO. Hoping this is fixed with new RC by some magic PR that touched account creation code. We'll see xD

**Comment by jasonbayton** [comment]
Sorry @mario I didn't see your account request :o 
You can always use the demo server as I can normally replicate there too?

**Comment by michaelof** [comment]
My 5 cents regarding account creation:

Switched from OwnCloud to NextCloud, so deinstalled OC android app, and installed current 1.4.2 F-Droid NextCloud android app.

Android 4.4.2

result: Nextcloud app crashes reproducible on start. 

Installed NextCloud "dev" app, version 20170903, also from F-Droid. Works like a charm :)

**Comment by AndyScherzinger** [comment]
For any store running Nextcloud 12+ you need to run at least 1.4.3 (not available on f-droid) or go for 2.0.0RC6.

**Comment by michaelof** [comment]
Somewhat weird...or maybe my lack of knowledge about F-Droid:

1.: for me, the nextcloud app crashes BEFORE I'm even able to select a server to connect to, so no chance for the 1.4.2, app to "detect" that my expected server to be connected might be 12.x

2. No idea why, maybe because if the * (star) in F-Droid's version history for nextcloud app, 1.4.2 is installed. Although much later, newer versions exists, e.g. 2.0.0RC5 and 2.0.0RC6.

Selected explicitly 2.0.0RC6 -> no crashes anymore

**Comment by AndyScherzinger** [comment]
1.4.2 had an issue on some phones which led to the app crashing right during start.

**Comment by AndyScherzinger** [comment]
fixed via 2.0.0
